lua/spectre_common.lua

```diff
@@ -60,19 +60,42 @@ local function get_vary_headers(headers, vary_headers_list)
     return table.concat(vary_headers, ',')
 end
 
-local function has_marker_headers(headers, marker_header_list)
+-- Helper function to check if available header value
+-- satisfies the comparison function for marker header lists.
+local function _check_headers_helper(headers, marker_header_list, compare_fn)
     for header, values in pairs(marker_header_list) do
         for _, v in pairs(values) do
             local lowercase_header_value = tostring(headers[header]):lower()
-            if lowercase_header_value == v then
+            if compare_fn(lowercase_header_value, v) then
                 return true
             end
         end
     end
-
     return false
 end
 
+local function has_marker_headers(headers, marker_header_list)
+    return _check_headers_helper(
+        headers,
+        marker_header_list,
+        function(actual, expected) return actual == expected end
+    )
+end
+
+-- Compares if the string starts with given substring
+local function string_starts_with(str, pattern)
+    return string.sub(str,1,string.len(pattern)) == pattern
+end
+
+-- checks if content-type headers are available.
+local function has_content_type_headers(headers, marker_header_list)
```

Comment:
This function has nothing "content_type" specific does it? Two solutions:
* make it content_type specific, which means you don't need `marker_header_list`, just a flat value (`has_content_type_header(headers, content_type_header_value)`)
* rename it to something generic: `headers_start_with` (not super happy about this name in particular but you get the idea)

Reply:
Updated method to signature `has_supported_content_type(headers)`